src/filters/network.rs

```diff
@@ -1351,11 +1361,17 @@ fn check_pattern_right_anchor_filter(filter: &NetworkFilter, request: &request::
 fn check_pattern_left_anchor_filter(filter: &NetworkFilter, request: &request::Request) -> bool {
     match &filter.filter {
         FilterPart::Empty => true,
-        FilterPart::Simple(f) => request.url.starts_with(f),
+        FilterPart::Simple(f) => if filter.match_case() {request.original_url.starts_with(f)} else {request.url.starts_with(f)},
         FilterPart::AnyOf(filters) => {
             for f in filters {
-                if request.url.starts_with(f) {
-                    return true;
+                if filter.match_case() {
+                    if request.original_url.starts_with(f) {
+                        return true;
+                    }
+                } else {
+                    if request.url.starts_with(f) {
+                        return true;
+                    }
```

Comment:
for patterns like this where both branches are the same except for the URL to check, it'd be easier to understand if you store a reference to the correct URL first, i.e.

```rust
let request_url = if filter.match_case() {
    &request.original_url
} else {
    &request.url
};
if request_url.starts_with(f) {
    return true;
}
```

Reply:
Thank you. Fixed.